service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/IdMapper.java

```diff
@@ -26,6 +26,10 @@ public String getOrNew(Reference reference) {
         return mappedId;
     }
 
+    public String getNew() {
```

Comment:
Since this id is not "mapped" by anything I think it would be better to leave this out of IdMapper. Just use RandomIdGeneratorService directly.